packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +121,53 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
+
+  const endpointUri = `${cloudHost}/user/shared`;
+  const getProductionData = () =>
+    fetch(endpointUri, {
+      credentials: 'include',
+      method: 'GET',
+    });
+
+  function getFixtureData() {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        onSuccess?.(fixtureData);
+
+        resolve(fixtureData);
+      }, 1000);
```

Comment:
I'd put this to maybe 10, just so it's not un-necessarily slow for consumers but still forces people to expect the component to load asyncronously.

Reply:
👍